pkg/certifier/scorecard/scorecard_test.go

```diff
@@ -0,0 +1,271 @@
+			wantErr:       true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if test.wantAuthToken {
+				t.Setenv("GITHUB_AUTH_TOKEN", test.authToken)
+			}
+
+			got, err := NewScorecard(test.sc, test.client)
+			if (err != nil) != test.wantErr {
+				t.Errorf("NewScorecard() error = %v, wantErr %v", err, test.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, test.want) {
+				t.Errorf("NewScorecard() got = %v, want %v", got, test.want)
+			}
+		})
+	}
+}
+
+func Test_scorecard_CertifyComponent(t *testing.T) {
+	type fields struct {
+		ghToken  string
+		artifact *assembler.ArtifactNode
+	}
+	type args struct {
+		ctx           context.Context
+		rootComponent interface{}
+		docChannel    chan<- *processor.Document
+	}
+	tests := []struct {
+		name                    string
+		fields                  fields
+		args                    args
+		getScoreShouldReturnErr bool
+		wantErr                 bool
+	}{
+		{
+			name: "doc chan is nil",
+			fields: fields{
+				ghToken:  "",
+				artifact: &assembler.ArtifactNode{},
+			},
+			args: args{
+				ctx:           context.Background(),
+				rootComponent: &assembler.ArtifactNode{},
+				docChannel:    nil,
+			},
+			wantErr: true,
+		},
+		{
+			name: "root component is nil",
+			fields: fields{
+				ghToken:  "",
+				artifact: &assembler.ArtifactNode{},
+			},
+			args: args{
+				ctx:           context.Background(),
+				docChannel:    make(chan *processor.Document),
+				rootComponent: nil,
+			},
+			wantErr: true,
+		},
+		{
+			name: "root component is not an artifact node",
+			fields: fields{
+				ghToken:  "",
+				artifact: &assembler.ArtifactNode{},
+			},
+			args: args{
+				ctx:           context.Background(),
+				docChannel:    make(chan *processor.Document),
+				rootComponent: "",
+			},
+			wantErr: true,
+		},
+		{
+			name: "scorecard getScore returns error",
+			fields: fields{
+				ghToken:  "",
+				artifact: &assembler.ArtifactNode{},
+			},
+			args: args{
+				ctx:           context.Background(),
+				docChannel:    make(chan *processor.Document),
+				rootComponent: &assembler.ArtifactNode{},
+			},
+			getScoreShouldReturnErr: true,
+			wantErr:                 true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			sc := mocks.NewMockScorecard(ctrl)
+			sc.EXPECT().GetScore(gomock.Any(), gomock.Any()).
+				DoAndReturn(func(a, b string) (*pkg.ScorecardResult, error) {
+					if test.getScoreShouldReturnErr {
+						return nil, fmt.Errorf("error")
+					}
+					return nil, nil
+				}).AnyTimes()
+
+			s := scorecard{
+				scorecard: sc,
+				ghToken:   test.fields.ghToken,
+			}
+			if err := s.CertifyComponent(test.args.ctx, test.args.rootComponent, test.args.docChannel); (err != nil) != test.wantErr {
+				t.Errorf("CertifyComponent() error = %v, wantErr %v", err, test.wantErr)
+			}
+		})
+	}
+}
+
+func TestCertifyComponent(t *testing.T) {
```

Comment:
I changed the names of the test functions so that `Test_scorecard_CertifyComponent` is now `Test_CertifyComponent` and `TestCertifyComponent` is now `TestCertifyComponentDefaultCase`.